b2dEditor.py

```diff
@@ -0,0 +1,391 @@
+    Button((button_x, button_y), (resolution[0] - button_x, button_y * 2.1), "dump_node_json()", "Export JSON")
+    Button((button_x, button_y), (resolution[0] - button_x, button_y * 3.2), "load_node_json()", "Import JSON")
+    Button((button_x, button_y), (resolution[0] - button_x, button_y * 4.3), "add_shape()", "Add Object")
+    Button((button_x, button_y), (resolution[0] - button_x, resolution[1] - button_y * 2), "exit()", "Exit")
+
+    # Main loop
+    while not quit:
+        # Limit FPS to 24 to minimize CPU usage
+        ticker.tick(24)
+
+        # Clear the screen
+        screen.fill((200, 200, 200))
+
+        mouse_pos = pygame.mouse.get_pos()
+
+        # Handle inputs
+        for event in pygame.event.get():
+
+            edit_area.handle_mouse(event)
+            for button in buttons:
+                button.handle_mouse(event)
+            if event.type == pygame.QUIT:
+                quit = True
+
+            elif event.type == pygame.KEYDOWN:
+                # Save if s is pressed
+                if event.key == pygame.K_s:
+                    dump_node_json()
+
+                # Load if l is pressed
+                elif event.key == pygame.K_l:
+                    load_node_json()
+
+                # Add shape if insert is pressed
+                elif event.key == pygame.K_INSERT:
+                    add_shape()
+
+        # Draw all buttons
+        for button in buttons:
+            pygame.draw.rect(screen, button.color, button.rect)
+            screen.blit(font.render(button.text, False, (200, 200, 200)), (button.rect[0] + 5, button.rect[1] + button.rect[3] / 3))
+
+        # Draw help text
+        pygame.draw.rect(screen, (50, 50, 50), ((resolution[0] - resolution[0] / 4, resolution[1] / 16 * 5.4), (resolution[0] / 4, font.get_height() * 2)))
+        screen.blit(font.render("Mouse1 : add node", False, (200, 200, 200)), (resolution[0] - resolution[0] / 4 + 5, resolution[1] / 16 * 5.4))
+        screen.blit(font.render("Mouse2 : delete node", False, (200, 200, 200)), (resolution[0] - resolution[0] / 4 + 5, resolution[1] / 16 * 5.4 + font.get_height()))
+
+        # Draw editing square
+        pygame.draw.rect(screen, edit_area.color, edit_area.rect)
+        pygame.draw.rect(screen, edit_area.edge_color, edit_area.rect, 2)
+
+        # Draw background image
+        if background:
+            screen.blit(background, ((0.5 * edit_area.rect.width) - (0.5 * background.get_rect().width) + 50, (0.5 * edit_area.rect.height) - (0.5 * background.get_rect().height) + 50))
+
+        if len(shapes) > 0:
+            # Iterate through shapes
+            for shape in range(len(shapes)):
+                nodes = shapes[shape]
+
+                # Set current shape color to green
+                if shape != current_shape:
+                    color = 1
+                else:
+                    color = 0
+
+                # Iterate through nodes and draw them
+                for i in range(len(nodes)):
```

Comment:
What about splitting this whole loop into a separate method, named, for instance, `draw_nodes()`?